### PR TITLE
Fix lint errors

### PR DESCRIPTION
**Pull request by lukebakken**
Start by removing some dead code.

@fho I have added you as a collaborator on my fork. Please feel free to push to this branch.

### CODE REVIEW

**Comment by lukebakken** [comment]
@fho I left some TODO statements to figure out what to do in some error situations.

**Comment by fho** [comment]
@lukebakken I'm looking into handling the error returned by `channel.recv`. 
It seems that `channel.recv()` never returns an error and that all the involved functions can be simplified to not return one.
If that works out, I'll create a separate PR to simplify that, otherwise I'll keep you uptodate :-)

**Comment by fho** [comment]
@lukebakken that worked, here is the PR: https://github.com/rabbitmq/amqp091-go/pull/80

**Comment by lukebakken** [comment]
Tests pass but there are still quite a few TODOs.

**Comment by lukebakken** [comment]
@fho @Zerpet @deadtrickster -

For now I think we can review this and address the `TODO`s around `panic` via #81. lmk what you think.

File: connection.go [hunk 1]
```diff
@@ -600,7 +614,14 @@ func (c *Connection) heartbeater(interval time.Duration, done chan *Error) {
 			// When reading, reset our side of the deadline, if we've negotiated one with
 			// a deadline that covers at least 2 server heartbeats
 			if interval > 0 {
-				conn.SetReadDeadline(time.Now().Add(maxServerHeartbeatsInFlight * interval))
+				if err := conn.SetReadDeadline(time.Now().Add(maxServerHeartbeatsInFlight * interval)); err != nil {
+					// TODO #78 what to do in error situation?
+					var op_error *net.OpError
+					if !errors.As(err, &op_error) {
```
Comment (ansd) [inline connection.go:620]: Would [errors.Is](https://pkg.go.dev/errors#Is) be better here given that variable `op_error` isn't used?

Comment (ansd) [reply connection.go:620]: In Go, use camel case instead of snake case.

**Review by ansd — CHANGES_REQUESTED** [review]

File: connection.go (hunk 1 above)
Comment (lukebakken) [reply connection.go:620]: I think we'd need an instance of `net.OpError` for `errors.Is` to work. I'll keep it as `.As` and will use the value.

**Review by ansd — APPROVED** [review]
As you already wrote, except for the `panic()` which should be avoided in libraries, the remaining changes look good to me. Thanks @lukebakken !

**Review by fho — COMMENTED** [review]
I think we should not add new `panic` calls.
Applications would now crash instead of ignoring errors in places where it seems to not have caused issues so far.

**Comment by lukebakken** [comment]
@fho we can do that.

One of the main reasons for this project to be created was that versioning was _not_ being used in https://github.com/streadway/amqp. If you're using `main` you should know the risks.

**Pull request merged**